apps/teams-test-app/src/public/auth_end.html

```diff
@@ -11,11 +11,16 @@ <h2 style="font-size: 40px;">Ending auth</h2>
       const { authId, method, hostName } = JSON.parse(params.get('state'));
 
       // Redirect back to Orange host
-      if (method === 'deeplink' && hostName === 'Orange') {
-        window.location.href = `msorange://testapp.com/auth_callback?authId=${authId}&result=${result}`;
-      } else {
-        document.getElementById('errorMessage').textContent = 'Unable to redirect back to the App';
+      if (method === 'deeplink') {
+        const storedRedirectUrl = sessionStorage.getItem('hostRedirectUrl')
+        if (storedRedirectUrl && storedRedirectUrl.trim() !== '') {
```

Comment:
>storedRedirectUrl.trim() !== '' [](http://example.com/codeflow?start=33&length=31)

Same question here about not using `trim()` and checking against empty string since that would be a bug that we would want it to fail on?